### PR TITLE
 CIS CNAME Setup : Create Partial Zone

### PR DESCRIPTION
**Pull request by arpit-srivastava-ibm**
CIS CNAME Setup - 
- Added Partial Zone creation option
- Added 2 new fields in response data for zone - `verification_key` & `cname_suffix`

### CODE REVIEW

**Comment by arpit-srivastava-ibm** [comment]
Added new Test case - `TestAccIBMCisPartialDomain_basic`. All test cases passed.
```
arpit-mac@Arpits-MacBook-Pro terraform-provider-ibm % make testacc TEST=./ibm/service/cis TESTARGS='-run=TestAccIBMCisPartialDomain_basic'

2022-07-28T15:58:54.041+0530 [DEBUG] sdk.helper_resource: Finished TestCase: test_name=TestAccIBMCisPartialDomain_basic
--- PASS: TestAccIBMCisPartialDomain_basic (298.22s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cis     299.095s
```

File: ibm/service/cis/data_source_ibm_cis_domain.go [hunk 1]
```diff
@@ -28,6 +28,12 @@ func DataSourceIBMCISDomain() *schema.Resource {
 				Description: "CISzone - Domain",
 				Required:    true,
 			},
+			cisDomainType: {
```
Comment (hkantare) [inline ibm/service/cis/data_source_ibm_cis_domain.go:31]: In datasource its a conputed not optional with default

File: website/docs/r/cis_domain.html.markdown [hunk 2]
```diff
@@ -40,6 +53,10 @@ In addition to all argument reference list, you can access the following attribu
 - `original_name_servers` - (String) The name servers that were used when the domain was first registered with the DNS Registrar.
 - `paused`- (Bool) Indicates if the domain is paused and network traffic bypasses your IBM Cloud Internet Services instance. The default values is **false**.
 - `status` - (String) The status of the domain. Valid values are `active`, `pending`, `initializing`, `moved`, `deleted`, and `deactivated`. After creation, the status remains pending until the DNS Registrar is updated with the CIS name servers, exported in the `name_servers` variable.
+- `type` - (String) The type of domain to be created. Default value is noted to be `full`- for regular domains, & to create a partial domain for CNAME setup, value to be used is `partial`.
```
Comment (hkantare) [inline website/docs/r/cis_domain.html.markdown:56]: Type is optional varaible it should be added under Argument Reference.
Add the doc changes for datasoruce also

File: ibm/service/cis/resource_ibm_cis_domain.go [hunk 3]
```diff
@@ -33,6 +36,12 @@ func ResourceIBMCISDomain() *schema.Resource {
 				Description: "CISzone - Domain",
 				Required:    true,
 			},
+			cisDomainType: {
+				Type:        schema.TypeString,
+				Description: "CISzone - Domain Type",
+				Default:     "full",
+				Optional:    true,
```
Comment (hkantare) [inline ibm/service/cis/resource_ibm_cis_domain.go:43]: Add validator funciton to allow only "full", "partial"

File: ibm/service/cis/data_source_ibm_cis_domain.go (hunk 1 above)
Comment (arpit-srivastava-ibm) [reply ibm/service/cis/data_source_ibm_cis_domain.go:31]: It should be Optional with default. It is not computed.

Comment (arpit-srivastava-ibm) [reply ibm/service/cis/data_source_ibm_cis_domain.go:31]: Corrected this

File: ibm/service/cis/resource_ibm_cis_domain.go (hunk 3 above)
Comment (arpit-srivastava-ibm) [reply ibm/service/cis/resource_ibm_cis_domain.go:45]: Added

File: examples/ibm-cis/myExmple/.terraform.lock.hcl [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# This file is maintained automatically by "terraform init".
```
Comment (hkantare) [inline examples/ibm-cis/myExmple/.terraform.lock.hcl:1]: you need this myExample folder also?

**Pull request merged**